OneCryptoPkg/Library/BaseCryptCrtLibOneCrypto/OneCryptoCrt.c

```diff
@@ -0,0 +1,156 @@
+/** @file
+  C Run-Time Libraries (CRT) implementations
+  These either call the shared implementations or implement the logic itself
+  if the implementation is simple enough that the code may be throughly vetted
+  to ensure safety.
+
+  Copyright (c) Microsoft Corporation. All rights reserved.
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+**/
+
+#include <Private/OneCryptoDependencySupport.h>
+#include <CrtLibSupport.h>
+
+/**
+  Allocates a buffer of a specified size from the pool.
+
+  This function allocates a buffer of size AllocationSize from the pool. If the
+  global shared dependencies or the AllocatePool function pointer within it is
+  NULL, the function returns NULL.
+
+  @param[in]  AllocationSize  The number of bytes to allocate.
+
+  @retval  NULL    If gOneCryptoDepends or gOneCryptoDepends->AllocatePool is NULL.
+  @retval  Others  A pointer to the allocated buffer.
+**/
+VOID *
+EFIAPI
+BaseCryptAllocatePool (
+  IN UINTN  AllocationSize
+  )
+{
+  ONE_CRYPTO_DEPENDENCIES  *Dependencies;
+
+  Dependencies = GetCryptoDependencies ();
+  if ((Dependencies == NULL) || (Dependencies->AllocatePool == NULL)) {
+    ASSERT (Dependencies != NULL);
+    ASSERT (Dependencies->AllocatePool != NULL);
+    return NULL;
+  }
+
+  return Dependencies->AllocatePool (AllocationSize);
+}
+
+/**
+  Allocates and zeros a buffer of a specified size from the pool.
+
+  This function allocates a buffer of size AllocationSize from the pool and then
+  zeros the entire allocated buffer. If the allocation fails, the function returns NULL.
+
+  @param[in]  AllocationSize  The number of bytes to allocate and zero.
+
+  @retval  NULL    If the allocation fails, or if AllocationSize is 0.
+  @retval  Others  A pointer to the allocated and zeroed buffer.
+**/
+VOID *
+EFIAPI
+BaseCryptAllocateZeroPool (
+  IN UINTN  AllocationSize
+  )
+{
+  VOID  *Buffer;
```

Comment:
Set buffer to NULL here because you don't know what the dependency will do. The std libs return null in failure, but can't guarantee with unknown dependency